src/coreclr/jit/llvm.cpp

```diff
@@ -1587,16 +1607,28 @@ void Llvm::buildReturn(GenTree* node)
 
 void Llvm::buildStoreInd(GenTreeStoreInd* storeIndOp)
 {
-    Value* address = getGenTreeValue(storeIndOp->Addr());
-    Value* toStore = getGenTreeValue(storeIndOp->Data());
-    if (toStore->getType()->isPointerTy() && (storeIndOp->gtFlags & GTF_IND_TGT_NOT_HEAP) == 0)
-    {
-        // RhpAssignRef will never reverse PInvoke, so do not need to store the shadow stack here
-        _builder.CreateCall(getOrCreateRhpAssignRef(), ArrayRef<Value*>{address, castIfNecessary(toStore, Type::getInt8PtrTy(_llvmContext))});
-    }
-    else
+    GenTree* data = storeIndOp->Data();
+    Value* toStore = getGenTreeValue(data);
+    Value* address = consumeValue(storeIndOp->Addr(), toStore->getType()->getPointerTo());
```

Comment:
```suggestion
  Value* address = consumeValue(storeIndOp->Addr(), getLlvmTypeForVarType(storeIndOp->TypeGet())->getPointerTo());
```
(Coupled with the comment about `data`)